File: backend/src/Accounting.Domain/Entities/LedgerTransaction.cs
using Accounting.Domain.Common;

namespace Accounting.Domain.Entities;

public class LedgerTransaction : BaseEntity
{
    public DateTime Date { get; set; }
    public long VoucherId { get; set; }
    public string VoucherNumber { get; set; } = string.Empty;
    public string VoucherType { get; set; } = "CP";

    public long AccountId { get; set; }
    public string Description { get; set; } = string.Empty;

    public decimal Debit { get; set; }
    public decimal Credit { get; set; }

    // Navigation Properties
    public virtual Voucher Voucher { get; set; } = null!;
    public virtual Account Account { get; set; } = null!;
}

File: app/src/main/java/com/example/domain/model/LedgerTransaction.kt
package com.example.domain.model

data class LedgerTransaction(
    val id: Long = 0,
    val date: Long,
    val voucherId: Long,
    val voucherNumber: String,
    val voucherType: String = "CP",
    val accountId: Long,
    val accountCode: String = "",
    val accountName: String = "",
    val accountType: AccountType = AccountType.ASSET,
    val description: String = "",
    val debitCents: Long = 0L,
    val creditCents: Long = 0L,
    val runningBalanceCents: Long = 0L
) {
    val runningBalance: Long
        get() = runningBalanceCents

    val netAmountCents: Long
        get() = debitCents - creditCents
}
